featuretools/feature_base/feature_base.py

```diff
@@ -103,6 +101,33 @@ def get_feature_names(self):
                     ]
         return self._names
 
+    def set_feature_names(self, names):
+        """Set new values for the feature column names, overriding the default values.
+        Number of names provided much match the number of output columns defined for
+        the feature. Only works for features that have more than one output column. Use
+        ``Feature.rename`` to change the column name for single output features.
+
+        Args:
+            names (list[str]): List of names to use for the output feature columns. Provided
+                names must be unique.
+        """
+        if self.number_output_features == 1:
+            raise ValueError(
+                "The set_feature_names can only be used on features that have more than one output column."
+            )
+
+        num_new_names = len(names)
+        if self.number_output_features != num_new_names:
+            raise ValueError(
+                "Number of names provided must match the number of output features:"
+                f" {num_new_names} name(s) provided, {self.number_output_features} expected."
+            )
+
+        if len(set(names)) != num_new_names:
+            raise ValueError("Provided output feature names must be unique.")
+
+        self._names = names
+
```

Comment:
An observation I have with this code base is that we are doing getters and setters like this:
```
class MyClass(object):
    def get_my_attr(self):
        return self._my_attr

    def set_my_attr(self, value):
        self._my_attr = value
```

When we really should be doing it the more pythonic way: 
```
class MyClass(object):
    @property
    def my_attr(self):
        return self._my_attr

    @my_attr.setter
    def my_attr(self, value):
        self._my_attr = value
```
anyway we could start using this pattern here? and maybe modify`get_feature_names` to just be

`@property`
`def feature_names` etc...

Reply:
Yes, I think `column_names` is better than `feature_names`.

Reply:
@ozzieD this is a great suggestion and definitely something we should do. Let's do this in a follow-up issue/PR so that we can get the release out today.

@thehomebrewnerd Can you create the issue?

Reply:
Yeah I have no issue with keeping it the way it is now and addressing it in a separate PR.

Reply:
https://github.com/alteryx/featuretools/issues/2161